nix/services/redis.nix

```diff
@@ -65,9 +89,22 @@ in
             {
               command = startScript;
 
-              readiness_probe = {
-                exec.command = "${config.package}/bin/redis-cli -p ${toString config.port} ping";
-              };
+              readiness_probe =
+                let
+                  # Transform `unixSocket` by prefixing `config.dataDir` if a relative path is used
+                  transformedSocketPath =
+                    if (config.unixSocket != null && (lib.hasPrefix "./" config.unixSocket)) then
```

Comment:
I think this doesn't work as intended, if I specify `unixSocket = "redis.socket";`, which is also a relative path, the socket will be created at `data/r1/redis.socket`. I think the check should be `! lib.hasPrefix "/" config.unixSocket`.

Reply:
https://github.com/juspay/services-flake/issues/365